packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useRef } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button
+   */
+  id: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled ("ON").
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. event 2. switch value
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  id,
+  labelId,
+  checked,
+  disabled,
+  onChange,
+  label,
+  ...customProps
+}) => {
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    sliderButtonRef.current.focus();
+    if (onChange) {
+      onChange(event, checked);
```

Comment:
I think we should prioritize these parameters with checked first to reduce unused params when implementing.

```jsx
onChange(checked, event);
```

Reply:
[updated](https://github.com/cerner/terra-core/pull/2846/commits/a30ab9bf52d60ad2b5b83dbe86defbb2a580b981#diff-ce03daf8e4362de3a438b2f322dedf6f)